core/types/cx_receipt.go

```diff
@@ -72,3 +76,106 @@ type CXMerkleProof struct {
 	ShardIDs      []uint32      // order list, records destination shardID
 	CXShardHashes []common.Hash // ordered hash list, each hash corresponds to one destination shard's receipts root hash
 }
+
+// CXReceiptsProof carrys the cross shard receipts and merkle proof
+type CXReceiptsProof struct {
+	Receipts    CXReceipts
+	MerkleProof *CXMerkleProof
+}
+
+// CXReceiptsProofs is a list of CXReceiptsProof
+type CXReceiptsProofs []*CXReceiptsProof
+
+// Len returns the length of s.
+func (cs CXReceiptsProofs) Len() int { return len(cs) }
+
+// Swap swaps the i'th and the j'th element in s.
+func (cs CXReceiptsProofs) Swap(i, j int) { cs[i], cs[j] = cs[j], cs[i] }
```

Comment:
What is this func for?

Reply:
It's not used, can be used to sort, here we might not needed it.